File: examples/go/s3_delete.go
// Elimina un objeto de S3 (y opcionalmente el bucket si queda vacio).
// Uso: go run s3_delete.go <bucket> <clave> [--bucket-tambien]
package main

import (
	"context"
	"fmt"
	"log"
	"os"

	"github.com/aws/aws-sdk-go-v2/aws"
	"github.com/aws/aws-sdk-go-v2/config"
	"github.com/aws/aws-sdk-go-v2/credentials"
	"github.com/aws/aws-sdk-go-v2/service/s3"
)

func main() {
	if len(os.Args) < 3 {
		log.Fatal("Uso: go run s3_delete.go <bucket> <clave> [--bucket-tambien]")
	}
	bucket := os.Args[1]
	key := os.Args[2]
	borrarBucket := len(os.Args) > 3 && os.Args[3] == "--bucket-tambien"

	ctx := context.Background()
	cfg, err := config.LoadDefaultConfig(ctx,
		config.WithRegion("us-east-1"),
		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider("test", "test", "")),
	)
	if err != nil {
		log.Fatalf("Error cargando configuracion: %v", err)
	}
	client := s3.NewFromConfig(cfg, func(o *s3.Options) {
		o.UsePathStyle = true
		o.BaseEndpoint = aws.String("http://localhost:4566")
	})

	if _, err := client.DeleteObject(ctx, &s3.DeleteObjectInput{Bucket: aws.String(bucket), Key: aws.String(key)}); err != nil {
		log.Fatalf("Error eliminando el objeto: %v", err)
	}
	fmt.Printf("Objeto eliminado: s3://%s/%s\n", bucket, key)

	if borrarBucket {
		if _, err := client.DeleteBucket(ctx, &s3.DeleteBucketInput{Bucket: aws.String(bucket)}); err != nil {
			log.Fatalf("Error eliminando el bucket (¿queda vacio?): %v", err)
		}
		fmt.Println("Bucket eliminado:", bucket)
	}
}

File: examples/rust/s3_delete.rs
// Elimina un objeto de S3 (y opcionalmente el bucket si queda vacio).
// Uso: cargo run --bin s3_delete -- <bucket> <clave> [--bucket-tambien]

use aws_config::BehaviorVersion;
use aws_credential_types::Credentials;
use aws_sdk_s3::{config::Region, Client, Config};

#[tokio::main]
async fn main() -> Result<(), aws_sdk_s3::Error> {
    let args: Vec<String> = std::env::args().collect();
    if args.len() < 3 {
        eprintln!("Uso: cargo run --bin s3_delete -- <bucket> <clave> [--bucket-tambien]");
        std::process::exit(1);
    }
    let bucket = &args[1];
    let key = &args[2];
    let borrar_bucket = args.get(3).map(|s| s.as_str()) == Some("--bucket-tambien");

    let config = Config::builder()
        .behavior_version(BehaviorVersion::latest())
        .endpoint_url("http://localhost:4566")
        .region(Region::new("us-east-1"))
        .credentials_provider(Credentials::new("test", "test", None, None, "floci"))
        .force_path_style(true)
        .build();
    let client = Client::from_conf(config);

    client.delete_object().bucket(bucket).key(key).send().await?;
    println!("Objeto eliminado: s3://{bucket}/{key}");

    if borrar_bucket {
        client.delete_bucket().bucket(bucket).send().await?;
        println!("Bucket eliminado: {bucket}");
    }
    Ok(())
}
